### DISCUSSION TITLE
3.1.0

### DISCUSSION THREAD

**Discussion by LabhanshAgrawal** [vercel/hyper#5753] [Announcements]
New stable release finally!!

|   OS    | Installer |                                                                                                 |                                                                                                  |
| :-----: | :-------: | :---------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------------------: |
|   Mac   |    dmg    | [x64](https://github.com/vercel/hyper/releases/download/v3.1.0/Hyper-3.1.0-mac-x64.dmg) (Intel) | [arm64](https://github.com/vercel/hyper/releases/download/v3.1.0/Hyper-3.1.0-mac-arm64.dmg) (M1) |
|  Linux  |    deb    |      [x64](https://github.com/vercel/hyper/releases/download/v3.1.0/hyper_3.1.0_amd64.deb)      |     [arm64](https://github.com/vercel/hyper/releases/download/v3.1.0/hyper_3.1.0_arm64.deb)      |
|         |    rpm    |     [x64](https://github.com/vercel/hyper/releases/download/v3.1.0/hyper-3.1.0.x86_64.rpm)      |    [arm64](https://github.com/vercel/hyper/releases/download/v3.1.0/hyper-3.1.0.aarch64.rpm)     |
|         | AppImage  |      [x64](https://github.com/vercel/hyper/releases/download/v3.1.0/Hyper-3.1.0.AppImage)       |   [arm64](https://github.com/vercel/hyper/releases/download/v3.1.0/Hyper-3.1.0-arm64.AppImage)   |
|         |   snap    |     [x64](https://github.com/vercel/hyper/releases/download/v3.1.0/hyper_3.1.0_amd64.snap)      |                                                                                                  |
| Windows |    exe    |      [x64](https://github.com/vercel/hyper/releases/download/v3.1.0/Hyper-Setup-3.1.0.exe)      |                                                                                                  |

<hr /><em>This discussion was created from the release <a href='https://github.com/vercel/hyper/releases/tag/v3.1.0'>3.1.0</a>.</em>

**Comment by Stanzilla** [comment]
First!

**Reply by meduzen** [reply]
Is there a comprehensive changelog somewhere?

**Reply by LabhanshAgrawal** [reply]
not yet

**Reply by leerob** [reply]
@meduzen I'm working on a blog post with the high-level updates!

**Reply by meduzen** [reply]
Thanks for the follow-up! 🙏

**Comment by corneliusroemer** [comment]
I updated and have some very weird characters appearing. Not sure whether this is because of the update or not, since I only started using 3.0.2 for a bit, then switched.

I use fish shell and fira code nf.

I downgraded to 3.0.2 again and it's fine now.

![image](https://user-images.githubusercontent.com/25161793/125995723-3620f4af-9da6-4304-9eb6-0e21f9edc889.png)

Edit: opened an issue since other people seem to have the same problem https://github.com/vercel/hyper/issues/5757

**Reply by bfaulk96** [reply]
I noticed this same thing

**Reply by vburojevic** [reply]
I have the same issue

**Reply by ghost** [reply]
I removed the font-ligatures plugin and this stopped. YMMV

**Comment by ArthurAlvesIF** [comment]
how to update whitout uninstall on linux?

**Reply by lefrog** [reply]
@ArthurAlvesIF do you know how you've installed it initially? Meaning snap, debian, rpm or AppImage?

**Reply by ArthurAlvesIF** [reply]
@leerob debian

**Reply by lefrog** [reply]
Download latest version
- go to that page and download the debian file or
- run the following command `wget https://github.com/vercel/hyper/releases/download/v3.1.1/hyper_3.1.1_amd64.deb`

To install the debian file run the following command
```
sudo dpkg --install hyper_3.1.1_amd64.deb
```
This automatically override previous installation with this newer version.

**Reply by ArthurAlvesIF** [reply]
Thanks!

**Comment by bfaulk96** [comment]
On top of the weird issue of letters rendering over other letters, I'm noticing some lag when typing that I didn't notice before updating.

**Comment by beargolightly** [comment]
I have updated and although I get a notification when I load Hyper that says "plugin error", it does nothing when I click on it (no extended message)
![image](https://user-images.githubusercontent.com/5595948/126009645-c9ea0a7b-a058-4c8e-98e1-41fd16e1e1bf.png)

When I use e.g. "hyper install hypercwd" I get:

"Response code 401 (Unauthorized)
Plugin check failed. Check your internet connection or retry later."

I'm not sure what the command to update all plugins might be or I would try it to see if I get the same message, but I reckon these two issues might have a common root cause 🤷‍♀️

Hyper launched without incident prior to the 3.1.0 update; **if there's a place to look for update/plugin logs I'd love to see it**, I searched in the ./hyper-plugins directory for interesting things but found nothing exciting.

Edit: it's worth noting that I originally installed Hyper using "arch -arm64 brew install -cask hyper" and Hyper auto-updated to 3.1.   There is no Cask for 3.1 so I can't try forcing a cask install/reinstall of 3.1 to see if that's the magic formula.

**Reply by LabhanshAgrawal** [reply]
are you using a private npm registry?

**Reply by beargolightly** [reply]
Solved:  

No private NPM registry, but I did solve the issue by deleting the old version with "arch -arm64 brew remove --cask hyper" and then installing it again with "arch -arm64 brew install --cask hyper"  (brew update did not work; after the upgrade it was still broken, had to remove and install)

**Comment by diegopeixoto** [comment]
@bfaulk96  i second that.
 so much lagging issues.

![](https://i.imgur.com/xTiKQF8.gif)

**Reply by LabhanshAgrawal** [reply]
Seems to be coming from the statusline plugin currently

**Reply by dellow** [reply]
Yea, hyperline is really slowing things down for me. If disabled everything is ok.

**Reply by EliasGcf** [reply]
Any idea why this lib is causing this?

**Reply by LabhanshAgrawal** [reply]
https://github.com/vercel/hyper/issues/5754#issuecomment-882761008

**Comment by a1ternat1ve** [comment]
It's very laggy and doesn't show text what I'm typing...

**Reply by bevanjkay** [reply]
If you are using hyperline try with it disabled.

**Comment by bevanjkay** [comment]
@LabhanshAgrawal Is it expected that having update channel set to Canary should update to a release when the release is newer than the canary?

**Reply by LabhanshAgrawal** [reply]
No
